### DISCUSSION TITLE
Fat models and skinny views?

### DISCUSSION THREAD

**Discussion by sebastian-philipp** [vitalik/django-ninja#459] [Q&A]
I'd like to get some advice from the experts here. Hopefully I'm not the only one with this question in mind. 

We need to write lots of trivial and not so trivial business logic and data validation somewhere in our code bases. Especially for the data validation, I'm a bit confused with django-ninja.

Having data validation in the API, has some implications:

* We're trusting that data that passes the API is consistent.
* If the API is bypassed, the models no longer validate their consistency, making it easy to corrupt the models.
* But is it a good idea to write Pydantic validators that execute database queries? 
* Having validations within the API layer typically give a better quality of error messages. 

On the other hand, my thoughts on fat models are:

* Pydantic makes it super convenient to put some good amount of validations into the API. We'd need to kinda replicate the very same logic in the django models again. 
* What about django model field validators? Is anyone using those?
* We'd end up with getters and setters for model fields that are unknown to the API layer, potentially bypassing things again. 

Finally, having a layer in between the API and models is also [questionable](https://www.b-list.org/weblog/2020/mar/16/no-service/)

Is there a canonical way to do things? My current approach is a mix of both, but that feels duplicated here and there.

**Comment by YAGregor** [comment]
I am not a expert, personally I think it's ok to query database during Pydantic running just like model serializer in  rest-framework.And no need to raise validate exception in view layer.
